### ISSUE TITLE
Calling generate_network() with ms_rewiring silently fails to rewire

### ISSUE THREAD

**Issue by ikbentimkramer** [marijnkeijzer/defSim#50]
A friend of mine is working on a simulation experiment where the network is being rewired. She tried to use some old code to gain some insight into what the rewired networks look like:
```python
import defSim as ds

G0 = ds.generate_network("connected_caveman_graph", l=3, k=3)
G1 = ds.generate_network("connected_caveman_graph", 
                         l=3, 
                         k=3, 
                         ms_rewiring=5/G0.number_of_edges())
```
However, as of #40, the `ms_rewiring` argument has been deprecated. In the case of `generate_network()` supplying `ms_rewiring` causes the function to silently fail to rewire, which caused quite some confusion.

In `initialize_simulation()` using `ms_rewiring` throws a warning:
https://github.com/marijnkeijzer/defSim/blob/81b618e42a168320e90f1712e2e01fa0b3f4bc57/defSim/Simulation.py#L177-L183

Perhaps implementing something similar in `generate_network()` is a good idea.

**Comment by dimaba** [comment]
Thanks Tim! Agreed, it would be best if the use of the ms_rewiring parameter is consistently treated this way. In addition, we should check whether mentions of ms_rewiring in the documentation have all been replaced.

Will fix this soon.

**Comment by dimaba** [comment]
This specific occurrence should be fixed in a504f35ae209326001c621d78111be4b2d515300
Leaving this open because we should check whether similar remains of ms_rewiring are left in other places, and should check whether documentation is up to date

**Comment by dimaba** [comment]
Can't find other instances where ms_rewiring is used without warning
